pilot/pkg/networking/util/util.go

```diff
@@ -587,6 +590,37 @@ func StringToPrefixMatch(in []string) []*matcher.StringMatcher {
 	return res
 }
 
+func ConvertToEnvoyMatches(in []*networking.StringMatch) []*matcher.StringMatcher {
+	res := make([]*matcher.StringMatcher, 0, len(in))
+
+	for _, im := range in {
+		if em := ConvertToEnvoyMatch(im); em != nil {
+			res = append(res, em)
+		}
+	}
+
+	return res
+}
+
+func ConvertToEnvoyMatch(in *networking.StringMatch) *matcher.StringMatcher {
```

Comment:
I would suggest keeping it in route package if it is only used there

Reply:
This is potentially useful for others as well similar to StringToExactMatch. That is why I kept here. Currently it is used only in route